immupdate.ts

```diff
@@ -253,7 +258,7 @@ class _Updater {
         const nextValue = clone(value)
         const newHost = isLast ? host : nextValue
 
-        host[this.data.field] = nextValue
+        if (nextValue !== undefined) host[this.data.field] = nextValue
```

Comment:
This line has been modified to pass this test :

```javascript
      // test if nothing happen when the element to delete is widely out of array's bound
      const updated3 = deepUpdate(data).at('a').at(5).set(DELETE)
      expect(updated3).toEqual(data)
```

Previous behaviour was to create the following object `{a: [1, 2, 3, undefined, undefined]}`

All unit tests pass, but I wonder, if it could introduced non tested regression.